packages/grafana-runtime/src/services/pluginExtensions/getPluginExtensions.ts

```diff
@@ -0,0 +1,25 @@
+import { PluginExtension } from '@grafana/data';
+
+export type GetPluginExtensions = ({ placement, context }: { placement: string; context?: object }) => {
+  extensions: PluginExtension[];
+};
+
+let singleton: GetPluginExtensions | undefined;
+
+export function setPluginExtensionGetter(instance: GetPluginExtensions): void {
+  // We allow overriding the registry in tests
+  if (singleton && process.env.NODE_ENV !== 'test') {
+    throw new Error('setPluginExtensionGetter() function should only be called once, when Grafana is starting.');
+  }
+  singleton = instance;
+}
+
+export function getPluginExtensionGetter(): GetPluginExtensions {
+  if (!singleton) {
+    throw new Error('getPluginExtensionGetter() can only be used after the Grafana instance has started.');
+  }
+  return singleton;
+}
```

Comment:
Do we need to expose this one if we are exposing the `getPluginExtensions`?

Reply:
No, that's a good point! 🙏